internal/source/git.go

```diff
@@ -0,0 +1,189 @@
+package source
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"strings"
+
+	"github.com/nlepage/go-tarfs"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	"github.com/operator-framework/rukpak/internal/git"
+	"github.com/operator-framework/rukpak/internal/util"
+)
+
+type Git struct {
+	Client          client.Client
+	KubeClient      kubernetes.Interface
+	ProvisionerName string
+	PodNamespace    string
+	UnpackImage     string
+	GitClientImage  string
+}
+
+const gitBundleUnpackContainerName = "bundle"
+
+func (g *Git) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Result, error) {
```

Comment:
What happens if I want my provisioner to unpack using a different mechanism than deploying and managing a Pod?

Reply:
That'd be fine, you'd pass a different implementation into `source.NewUnpacker` that does something else in this function.